src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i].type;
+
+    if (action === 'clear') {
+      for (const stateKey in state) {
+        delete state[stateKey];
+      }
+    } else if (action === 'addProperties') {
+      const addObject = actions[i].extraData;
+
+      for (const addKey in addObject) {
+        state[addKey] = addObject[addKey];
+      }
+    } else if (action === 'removeProperties') {
```

Comment:
Use the correct action property name. Replace `else if (action === 'removeProperties')` with `else if (action.type === 'removeProperties')`

Reply:
fixed